rust/src/writer/stats.rs

```diff
@@ -238,8 +239,16 @@ impl StatsScalar {
                     });
                 };
 
-                let val = val / 10.0_f64.powi(*scale);
-                Ok(Self::Decimal(val))
+
+                let str_val = val.to_string();
+                let decimal_string = if str_val.len() > *scale as usize {
+                    let (integer_part, fractional_part) =
+                        str_val.split_at(str_val.len() - *scale as usize);
+                    format!("{}.{}", integer_part, fractional_part)
+                } else {
+                    format!("0.{}", str_val)
+                };
```

Comment:
I think we also want to have negative scales. Also, could we unit test this directly in this file?

Reply:
Added support for negative scales. Not going to be able to add tests for a bit. I added you to my fork so you can commit if you want to take a crack at it.